### PR TITLE
Issue 19 Resolve hosts permissions issues on submission

### LINKED ISSUE #19: Alternate option to using "host" permission
When attempting to submit to the chrome store, the store gave strong language that we're going to have to wait a long time for review because of the permissions we are requesting. Specifically the "host" permission. From the verbage, it sounds like there is an alternative way that won't make our submission get delayed

Screenshot 2024-08-06 at 9 00 39 PM

### PR DESCRIPTION
**Pull request by Tanner-Scadden**
## The Pull Request is ready

- [ ] all github actions are passing
- [ ] only a single issue was worked on
- [ ] fixes #19
- [ ] the branch follows the naming schema `issue-123-enable-x-does-not-disable-y`
- [ ] the pull request has a sensible title

## Intention

With this change I intend to...
- Remove any hosts permissions
- Added localhost as dev site
- Change it back to all tabs since activeTab is being weird and not grabbing URL. Probably need to fix that and adjust.

## Review Points

Please take extra care reviewing...

## The code follows best practices

- [ ] duplicate code has been extracted where possible
- [ ] issues for follow-up tasks have been created
- [ ] tests have been written for any new functionality
- [ ] there is no `any` type used
- [ ] texts have been checked for grammar and spelling issues

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -23,10 +23,17 @@
     "service_worker": "src/routes/background/background.ts",
     "type": "module"
   },
-  "permissions": ["alarms", "tabs", "storage", "idle"],
+  "permissions": ["idle", "storage", "alarms", "tabs"],
+  "optional_host_permissions": ["https://*/*", "http://*/*"],
   "content_scripts": [
     {
-      "matches": ["http://*/*", "https://*/*", "<all_urls>"],
```
Comment (Tanner-Scadden) [inline manifest.json:29]: [This article explains why we got the error](https://developer.chrome.com/docs/extensions/develop/concepts/declare-permissions#host-permissions)

**Review by rphovley — DISMISSED** [review]

File: manifest.json (hunk 1 above)
Comment (rphovley) [reply manifest.json:29]: Unfortunately, with this change to make the match patterns more specific, the host permissions is still there

Comment (rphovley) [reply manifest.json:29]: It is this 'matches' argument though. When I remove those from the manifest the host permission drops

Comment (Tanner-Scadden) [reply manifest.json:29]: I thought I updated this PR not to have the content_scripts anymore. Sorry!

**Review by rphovley — APPROVED** [review]

**Pull request merged**